### ISSUE TITLE
Errors building with Visual Studio 2019

### ISSUE THREAD

**Issue by nashmuhandes** [yquake2/yquake2#981]
Hi, I've followed the build instructions from https://github.com/yquake2/yquake2/blob/master/doc/020_installation.md, tried build both a 32 bit and 64 bit - both have a ton of errors.

Here are some of the error messages:

```
5>C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\um\oaidl.h(487,17): error C2059: syntax error: '/'
5>C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\um\oaidl.h(531,13): error C2059: syntax error: '}'
6>C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\um\oaidl.h(487,17): error C2059: syntax error: '/' (compiling source file D:\Projects\Game Projects\Quake 2\yamagi\yquake2\src\client\refresh\gl1\gl1_scrap.c)
5>C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\um\oaidl.h(534,5): error C2059: syntax error: '}'
```

There's a whole lot more, there are hundreds of these littered in the output window.

I have also made sure that `dhewm3-libs` is properly set up, as per the instructions in the docs.

**Comment by nashmuhandes** [comment]
Here's what I use to generate the solution:

```
cmake -G "Visual Studio 16 2019" -A Win32 -DYQUAKE2LIBS="D:/Projects/Game Projects/Doom3/dhewm3-libs/i686-w64-mingw32" "D:/Projects/Game Projects/Quake2/yamagi/yquake2"
pause
```

**Comment by nashmuhandes** [comment]
Here are full outputs of the CMake and the Release Build Solution.

[CMake Output.txt](https://github.com/yquake2/yquake2/files/11018146/CMake.Output.txt)
[Release Build Output.txt](https://github.com/yquake2/yquake2/files/11018149/Release.Build.Output.txt)

**Comment by turol** [comment]
What's on the failing lines in `oaidl.h`? Probably some macro defined by yquake2 is causing problems for that header.

**Comment by nashmuhandes** [comment]
> What's on the failing lines in `oaidl.h`? Probably some macro defined by yquake2 is causing problems for that header.

Double-clicking it on the output list in VS2019 brings me here:

![image](https://user-images.githubusercontent.com/4926156/226353683-6cf02256-5e14-4185-a8fb-e3d599aeab0c.png)

**Comment by nashmuhandes** [comment]
Sorry for the spam; expanding `_VARIANT_BOOL` shows me this:

![image](https://user-images.githubusercontent.com/4926156/226354128-e8c07387-df63-4e94-81a0-24ffd5d6df16.png)

**Comment by turol** [comment]
This is some MSVC weirdness, we define neither `_VARIANT_BOOL` nor `bool`. For some reason it doesn't like the include order.

**Comment by turol** [comment]
Does the compiler tell you which headers it's trying to include when it gets to that file? It might be possible to work around this by changing include order.

**Comment by nashmuhandes** [comment]
Hi, some progress.

I've updated my Windows SDK to Windows 11 SDK (version `10.0.22000.0`) and the errors I'm getting now are much less (whew) - only the following:

```
2>D:\Projects\Game Projects\Quake2\yamagi\yquake2\src\client\curl\download.c(86,66): error C2065: 'CURLINFO_CONTENT_LENGTH_DOWNLOAD_T': undeclared identifier
2>D:\Projects\Game Projects\Quake2\yamagi\yquake2\src\client\curl\download.c(302,46): error C2065: 'CURLOPT_XFERINFOFUNCTION': undeclared identifier
```

I have made sure that the dhewm3 libraries are properly set up.

**Comment by Yamagi** [comment]
The *dhewm3-libs* ship with a more than ancient curl 8.23.1. That  version was released more than 12 years ago. Since e17d4f5dc0f209214d148f4335a9280ebaab5b97 YQ2 has required at least curl 7.50. You ca either revert that revision or you need to provide an up to date curl.

**Comment by Yamagi** [comment]
Arg, typo. curl 7.23.1 of course.

**Comment by DanielGibson** [comment]
I'll try to remember to update dhewm3libs soon

**Comment by nashmuhandes** [comment]
Thank you. That was it. I chose the easier option and just reverted commit https://github.com/yquake2/yquake2/commit/e17d4f5dc0f209214d148f4335a9280ebaab5b97 for the time being.

I now have compiled executables and the game DLL sitting in my Release folder.

I'm too incompetent to figure out how to update the curl that's sitting inside the `dhewm3libs` folder so I'll just wait for DanielGibson to do that in future. :)

So, tl;dr, for anyone else having trouble with Visual Studio:

- Update your Windows SDK (I updated mine to Windows 11 SDK 10.0.22000.0)
- Update libcurl inside dhewm3libs -OR- revert commit https://github.com/yquake2/yquake2/commit/e17d4f5dc0f209214d148f4335a9280ebaab5b97 in your fork

**Comment by DanielGibson** [comment]
I just updated dhewm3-libs with a recent libcurl

and FWIW, for me the build with VS2019 and "Windows SDK version 10.0.19041.0 to target Windows 10.0.19045" worked just fine (it seems to be a bit more recent than the one you originally tried, but at least it's not for Win11)

**Comment by nashmuhandes** [comment]
> I just updated dhewm3-libs with a recent libcurl
> 
> and FWIW, for me the build with VS2019 and "Windows SDK version 10.0.19041.0 to target Windows 10.0.19045" worked just fine (it seems to be a bit more recent than the one you originally tried, but at least it's not for Win11)

Noted on both accounts - thanks. Yeah, seemed overkill to me too, to update to the Win11 SDK. Didn't sit right with me. I'll downgrade to the version you mentioned.
